### ISSUE TITLE
Dexonline Office Add-in

### ISSUE THREAD

**Issue by andreiepure** [dexonline/dexonline#548]
Mă gândesc undeva în viitor[ii ani] să creez extensii pentru Office folosing API-ul dexonline. Cred că e bine să împart ideea cu comunitatea, pentru a discuta și poate le face altcineva sau le facem colaborativ.

Office Add-ins [1] sunt aplicații web (html, css, javascript) găzduite într-un iframe în Office. Printr-un API javascript, se poate accesa conținutul documentului sau a tabelelor Excel etc.

Funcționalități posibile pentru Word (cărți, referate, articole), Outlook (meiluri), PowerPoint (prezentări la școală), OneNote (jurnal de bord):
- click dreapta pe un cuvânt  pentru a căuta definiția
- căutat definiții/conjugări direct din contextul word
- inserat definiția unui cuvânt în document (poate fi util pentru note infrapaginale, de exemplu)

Întrebări pentru comunitate:
- Vede cineva vreun folos pentru Excel?
- Aveți alte idei de funcționalități?
- Există aplicații existente open source care folosesc API-ul dexonline? (links please)

Mulțumesc
Andrei 

[1] https://dev.office.com/docs/add-ins/overview/office-add-ins

**Comment by CatalinFrancu** [comment]
Noi nu prea vrem să investim timp în software neliber. Și pe IE testăm din an în Paște. :-) Dar dacă comunitatea vrea, sigur, sunteți bineveniți. Poate ne alegem și noi în sfârșit cu un API!

**Comment by revolter** [comment]
Deci exista sau nu un API momentan?

**Comment by CatalinFrancu** [comment]
Scuze pentru neclaritate. Nu există un API. Implementarea acestui tichet ar depinde de crearea unui API.

**Comment by danburzo** [comment]
Am putea deschide un issue nou pentru a aduna idei despre API. Site-ul actual este în sine un soi de API — poți să ceri definițiile oricărui cuvînt cerînd pagina `http://dexonline/definitie/<cuvant>` — deci se poate construi și pe baza asta. Două idei, una mai simplă decît cealaltă, dar ambele mai simple decît un API separat, ar fi:

1.  să anotăm pagina HTML într-un format de genul [microdata](https://schema.org/), astfel încît să se poată extrage informația printr-un parser de DOM, iar request-urile se pot face la paginile normale
2. să introducem un _query parameter_ pentru format — de exemplu, `http://dexonline.ro/definitie/<cuvant>?format=json`, care să sară peste părțile de templating și să servească conținutul brut în formatul respectiv ([pe modelul Wikimedia](https://ro.wikipedia.org/w/api.php?action=parse&page=DEX%20online&format=json))

**Comment by revolter** [comment]
> Scuze pentru neclaritate.

Nu-i nimic, doar ca @andreiepure a zis "folosing API-ul dexonline" si a parut ca exista unul.

> poți să ceri definițiile oricărui cuvînt cerînd pagina `http://dexonline/definitie/<cuvant>`

Asta am si facut momentan, dar asta e crawling deja, si trebuie parsat HTML-ul si nu e asa usor.

> 1. să anotăm pagina HTML într-un format de genul microdata, astfel încît să se poată extrage informația printr-un parser de DOM, iar request-urile se pot face la paginile normale

Asta are dezavantajul ca downloadezi si continut de care nu ai nevoie (restul paginii).

> 2. să introducem un query parameter pentru format — de exemplu, http://dexonline.ro/definitie/<cuvant>?format=json, care să sară peste părțile de templating și să servească conținutul brut în formatul respectiv (pe modelul Wikimedia)

Asta suna interesant. Dar, informatiile vin direct din MySQL sau dintr-un Redis sau alt sistem de caching?

**Comment by danburzo** [comment]
> informatiile vin direct din MySQL sau dintr-un Redis sau alt sistem de caching?

La întrebarea asta nu știu să-ți răspund momentan; îmi închipui dacă substitui randarea HTML cu una JSON, va beneficia de același nivel de caching.

**Comment by danburzo** [comment]
Am început un thread nou pe #562

**Comment by revolter** [comment]
> îmi închipui dacă substitui randarea HTML cu una JSON, va beneficia de același nivel de caching.

Asta daca exista deja caching pentru varianta HTML.
